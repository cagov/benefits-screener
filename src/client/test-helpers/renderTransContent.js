import React from "react";
import { mount } from "enzyme";
import { I18nextProvider } from "react-i18next";
import i18n from "./i18nForTests";

function renderTransContent(Component, props = {}) {
  const wrapper = mount(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  // Filter out the very verbose I18nextProvider component wrapper
  return wrapper.find(Component);
}

module.exports = renderTransContent;
