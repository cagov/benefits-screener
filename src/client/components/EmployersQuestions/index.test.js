import renderTransContent from "../../test-helpers/renderTransContent";
import Component from "./index";

describe("<EmployersQuestions />", () => {
  it("renders the component", async () => {
    const wrapper = renderTransContent(Component, {
      onChange: () => {},
    });

    expect(wrapper).toMatchSnapshot();
  });

  it("renders with 2 income sources", async () => {
    const wrapper = renderTransContent(Component, {
      onChange: () => {},
      employers: [
        {
          id: "12345678",
          employerName: "employer name 1",
          address1: "e1 - address 1",
          address2: "",
          city: "city",
          state: "CA",
          zipcode: "94100",
          lastDateWorked: "02/22/2020",
          totalHoursWorked: "10",
          grossEarnings: "$200",
          reason: "still-working",
          moreDetails: "",
        },
        {
          id: "22222222",
          employerName: "employer name 2",
          address1: "e2 - address 1",
          address2: "",
          city: "city 2",
          state: "OR",
          zipcode: "97035",
          lastDateWorked: "02/24/2020",
          totalHoursWorked: "12",
          grossEarnings: "$300",
          reason: "quit",
          moreDetails: "unable to travel",
        },
      ],
    });

    expect(wrapper).toMatchSnapshot();
  });
});
