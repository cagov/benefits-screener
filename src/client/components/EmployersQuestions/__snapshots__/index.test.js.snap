// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployersQuestions /> renders the component 1`] = `
<div>
  <p>
    Tell us about your income.
  </p>
  <div
    key="employerData00000000"
  >
    <hr />
    <Row
      className="align-items-baseline"
      noGutters={false}
    >
      <Col
        md="auto"
      >
        <h3>
          <Trans
            i18nKey="retrocerts-certification.employers.heading"
            t={[Function]}
            values={
              Object {
                "incomeNumber": 1,
              }
            }
          />
        </h3>
      </Col>
    </Row>
    <EmployerQuestions
      employerData={
        Object {
          "address1": "",
          "address2": "",
          "city": "",
          "employerName": "",
          "grossEarnings": "",
          "id": "00000000",
          "lastDateWorked": "",
          "moreDetails": "",
          "reason": "still-working",
          "state": "",
          "totalHoursWorked": "",
          "zipcode": "",
        }
      }
      onChange={[Function]}
    />
  </div>
  <hr />
  <Row
    noGutters={false}
  >
    <Col
      md="auto"
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        onClick={[Function]}
      >
        + Add another source of income
      </NavLink>
    </Col>
  </Row>
</div>
`;

exports[`<EmployersQuestions /> renders with 2 income sources 1`] = `
<div>
  <p>
    Tell us about your income.
  </p>
  <div
    key="employerData12345678"
  >
    <hr />
    <Row
      className="align-items-baseline"
      noGutters={false}
    >
      <Col
        md="auto"
      >
        <h3>
          <Trans
            i18nKey="retrocerts-certification.employers.heading"
            t={[Function]}
            values={
              Object {
                "incomeNumber": 1,
              }
            }
          />
        </h3>
      </Col>
      <Col
        md="auto"
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
        >
          Remove this source of income
        </NavLink>
      </Col>
    </Row>
    <EmployerQuestions
      employerData={
        Object {
          "address1": "e1 - address 1",
          "address2": "",
          "city": "city",
          "employerName": "employer name 1",
          "grossEarnings": "$200",
          "id": "12345678",
          "lastDateWorked": "02/22/2020",
          "moreDetails": "",
          "reason": "still-working",
          "state": "CA",
          "totalHoursWorked": "10",
          "zipcode": "94100",
        }
      }
      onChange={[Function]}
    />
  </div>
  <div
    key="employerData22222222"
  >
    <hr />
    <Row
      className="align-items-baseline"
      noGutters={false}
    >
      <Col
        md="auto"
      >
        <h3>
          <Trans
            i18nKey="retrocerts-certification.employers.heading"
            t={[Function]}
            values={
              Object {
                "incomeNumber": 2,
              }
            }
          />
        </h3>
      </Col>
      <Col
        md="auto"
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
        >
          Remove this source of income
        </NavLink>
      </Col>
    </Row>
    <EmployerQuestions
      employerData={
        Object {
          "address1": "e2 - address 1",
          "address2": "",
          "city": "city 2",
          "employerName": "employer name 2",
          "grossEarnings": "$300",
          "id": "22222222",
          "lastDateWorked": "02/24/2020",
          "moreDetails": "unable to travel",
          "reason": "quit",
          "state": "OR",
          "totalHoursWorked": "12",
          "zipcode": "97035",
        }
      }
      onChange={[Function]}
    />
  </div>
  <hr />
  <Row
    noGutters={false}
  >
    <Col
      md="auto"
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        onClick={[Function]}
      >
        + Add another source of income
      </NavLink>
    </Col>
  </Row>
</div>
`;
